sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/BuilderHelper.java

```diff
@@ -49,15 +50,22 @@
 
     static HttpPipeline buildPipeline(AzureNamedKeyCredential azureNamedKeyCredential,
                                       AzureSasCredential azureSasCredential, TokenCredential tokenCredential,
-                                      String sasToken, String endpoint, RetryPolicy retryPolicy,
+                                      String sasToken, String endpoint,
+                                      RetryPolicy retryPolicy, RetryOptions retryOptions,
                                       HttpLogOptions logOptions, ClientOptions clientOptions, HttpClient httpClient,
                                       List<HttpPipelinePolicy> perCallAdditionalPolicies,
                                       List<HttpPipelinePolicy> perRetryAdditionalPolicies, Configuration configuration,
                                       ClientLogger logger) {
         configuration = (configuration == null) ? Configuration.getGlobalConfiguration() : configuration;
-        retryPolicy = (retryPolicy == null) ? new RetryPolicy() : retryPolicy;
         logOptions = (logOptions == null) ? new HttpLogOptions() : logOptions;
 
+        if (retryPolicy != null && retryOptions != null) {
```

Comment:
We should make sure methods that use `buildPipeline()` include a `@throws` section talking about this case in their Javadoc.